modules/runner/src/integrationTest/java/org/apache/ignite/internal/table/ItTableScanTest.java

```diff
@@ -336,6 +337,36 @@ public void testGetAndDeleteDuringPureTableScan() throws Exception {
         );
     }
 
+    /**
+     * Ensures that multiple consecutive scan requests with different requested rows amount
+     * return the expected total number of requested rows.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testCompositeScanRequest() throws Exception {
+        int[][] demandPairs = {{3, 1}, {1, 3}};
+
+        for (int[] input : demandPairs) {
+            List<BinaryRow> scannedRows = new ArrayList<>();
+            Publisher<BinaryRow> publisher = internalTable.scan(0, null, null, null, null, 0, null);
+            CompletableFuture<Void> scanned = new CompletableFuture<>();
+
+            Subscription subscription = subscribeToPublisher(scannedRows, publisher, scanned);
+
+            subscription.request(input[0]);
+            subscription.request(input[1]);
+
+            int total = input[0] + input[1];
+            waitForCondition(() -> scannedRows.size() == total, 10_000);
+            assertEquals(total, scannedRows.size(), "input=" + Arrays.toString(input));
```

Comment:
we already check rows in previous line, isn`t it ?

Reply:
In the previous line, we just wait for a condition. The condition may fail.
But `assertTrue(waitForCondition(...))` will not display the actual number of rows received. Thus, I prefer to do the final check with an explicit `assertEquals(...)`. If you insist on adding `assertTrue` I can redo it.

Reply:
Reworked to single line condition.